src/Pages/Detail/index.js

```diff
@@ -0,0 +1,258 @@
+import React, { useState, useEffect, useRef } from "react";
+import styled from "styled-components";
+import axios from "axios";
+import Aside from "./Aside";
+import Banner from "./Components/Banner";
+import Review from "./Components/Review";
+import LearningGoal from "./Components/LearningGoal";
+import WhatIsClass from "./Components/WhatIsClass";
+import Community from "./Components/Community";
+import HeadModal from "./Components/HeadModal";
+import { Hr } from "./Components/Utils";
+import { AiFillExclamationCircle } from "react-icons/ai";
+import TopHeader from "./Components/TopHeader";
+import { GetClassInfo } from "../../store/DetailReducer";
+import { useSelector, useDispatch } from "react-redux";
+import { API } from "./Components/Utils";
+
+const Detail = () => {
+  const [isTopModal, setIsTopModal] = useState(false);
+  const [scrollValue, setScrollValue] = useState(0);
+  const [selectTab, setSelectTab] = useState(0);
+  const [slickWidth, setSlickWidth] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const { Detail, HeaderImages } = useSelector((store) => store.DetailReducer);
+  const dispatch = useDispatch();
+  const focusTarget = useRef([]);
+  const scrollEvent = () => {
+    setScrollValue(document.documentElement.scrollTop);
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", scrollEvent);
+    axios.get(API).then((res) => {
+      dispatch(GetClassInfo(res.data.detail));
+    });
+    return () => {
+      window.removeEventListener("scroll", scrollEvent);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (scrollValue < focusTarget.current[1].offsetTop - 50) {
+      setSelectTab(1);
+    } else {
+      setSelectTab(2);
+    }
+  }, [scrollValue]);
```

Comment:
```suggestion
  useEffect(() => {
    const isFirstTabActive = scrollValue < focusTarget.current[1].offsetTop - 50
    setSelectTab(isFirstTabActive ? 1 : 2)
  }, [scrollValue]);
```
삼항연산자를 사용하여 위와 같이 사용할 수 있습니다